packages/payload/src/fields/validations.ts

```diff
@@ -640,7 +645,8 @@ export const upload: UploadFieldValidation = async (value, options) => {
     }
   }
 
-  if (typeof value !== 'undefined' && value !== null) {
+  // Only run this validation for mongodb
+  if (typeof value !== 'undefined' && value !== null && payload?.db?.name === 'mongoose') {
```

Comment:
Why would you only do this in mongodb?
The filteroptions is doing more than just ensuring reference docs exist. The main goal is to enforce application logic and validate the related doc meets some conditions.